packages/flutter/lib/src/rendering/sliver_group.dart

```diff
@@ -0,0 +1,141 @@
+/// it is assumed to have a flex value of 1, which by itself expands
+/// to take up all of the cross axis extent.
+class RenderSliverCrossAxisGroup extends RenderSliver with ContainerRenderObjectMixin<RenderSliver, SliverPhysicalContainerParentData> {
+  @override
+  void setupParentData(RenderObject child) {
+    if (child.parentData is! SliverPhysicalContainerParentData) {
+      child.parentData = SliverPhysicalContainerParentData();
+    }
+  }
+
+  @override
+  double childMainAxisPosition(RenderSliver child) => 0.0;
+
+  @override
+  double childCrossAxisPosition(RenderSliver child) {
+    switch (constraints.axisDirection) {
+      case AxisDirection.up:
+      case AxisDirection.down:
+        return (child.parentData! as SliverPhysicalContainerParentData).paintOffset.dx;
+      case AxisDirection.left:
+      case AxisDirection.right:
+        return (child.parentData! as SliverPhysicalContainerParentData).paintOffset.dy;
+    }
+  }
+
+  @override
+  void performLayout() {
+    // Iterate through each sliver.
+    // Get the parent's dimensions.
+    final double crossAxisExtent = constraints.crossAxisExtent;
+    assert(crossAxisExtent.isFinite);
+
+    // First, layout each child with flex == 0.
+    int totalFlex = 0;
+    double usedUpExtent = 0.0;
+    RenderSliver? child = firstChild;
+    while (child != null) {
+      final SliverPhysicalContainerParentData childParentData = child.parentData! as SliverPhysicalContainerParentData;
+      final int flex = childParentData.flex ?? 1;
+      if (flex == 0) {
+        child.layout(constraints, parentUsesSize: true);
+        usedUpExtent += child.geometry!.crossAxisExtent;
+        print(usedUpExtent);
+      } else {
+        totalFlex += flex;
+      }
+      child = childAfter(child);
+    }
+    final double extentPerFlexValue = (crossAxisExtent - usedUpExtent) / totalFlex;
+
+    child = firstChild;
+    double offset = 0.0;
+
+    // At this point, all slivers with constrained cross axis should already be laid out.
+    geometry = SliverGeometry.zero;
+    while (child != null) {
+      final SliverPhysicalContainerParentData childParentData = child.parentData! as SliverPhysicalContainerParentData;
+      final int flex = childParentData.flex ?? 1;
+      double childExtent;
+      if (flex != 0) {
+        childExtent = extentPerFlexValue * flex;
+        child.layout(constraints.copyWith(
+          crossAxisExtent: extentPerFlexValue * flex,
+        ), parentUsesSize: true);
+      } else {
+        childExtent = child.geometry!.crossAxisExtent;
+      }
+      // Set child parent data.
+      switch (constraints.axis) {
+        case Axis.vertical:
+          childParentData.paintOffset = Offset(offset, 0.0);
+        case Axis.horizontal:
+          childParentData.paintOffset = Offset(0.0, offset);
+      }
+      offset += childExtent;
+      if (geometry!.scrollExtent < child.geometry!.scrollExtent) {
+        geometry = child.geometry;
```

Comment:
The geometry of the group would not have the same geometry of one of the children, after all, the group must have the geometry of the full cross axis extent, not just the cross axis extent of one child. :)

Reply:
Fixed! I copied the child geometry with the correct crossAxisExtent at the end.